### ISSUE TITLE
Router load attribute: broken when using id & parameters.bind

### ISSUE THREAD

**Issue by migajek** [aurelia/aurelia#1902]
**tl;dr** `load` attribute broken when using `id` and `parameters.bind`: 
`Uncaught (in promise) Error: 1 remaining instructions after 100 iterations; there is likely an infinite loop.`

I started with the original "recipes" app from router tutorial. Modified the stackblitz only to update aurelia 2 to beta10.
Then I've: 
- added `id: 'recipe-details',` to Recipe Details route definition
- modified recipes-page template to following:
```
<a load="/recipes/${recipe.idMeal}">${recipe.strMeal} [original]</a> |
    <a load="id: recipe-details; parameters.bind: {recipeId: recipe.idMeal}"><u>${recipe.strMeal} [bug]</u></a>
```
- both elements render identical `a href` (see below):
 
![image](https://github.com/aurelia/aurelia/assets/581492/ee94ac18-69e0-46c2-a938-4e2ecc0ee421)

- however when clicking the first one (original) it loads fine. when clicking the second one, I'm getting
`Uncaught (in promise) Error: 1 remaining instructions after 100 iterations; there is likely an infinite loop.`

**Repro**: please head to "recipes" page and try opening any of the recipe details *using the underlined link*. See console for error.
https://stackblitz.com/edit/au2-conventions-ptqpuz?file=src%2Frecipes-page.html

**Comment by johnmanko** [comment]
@migajek 

Have you tried changing `parameters.bind` to `params.bind`?  

From the [docs](https://docs.aurelia.io/getting-to-know-aurelia/routing/aurelia-router/navigating#html-load-attribute):

> And where things start to get interesting is when you want to pass parameters to a route. We use the params configuration property to specify parameters.

**Comment by migajek** [comment]
@johnmanko unfortunately the docs is simply wrong on this, as stated in the comment here by @jwx :
https://github.com/aurelia/aurelia/issues/1835#issuecomment-1892845707

> In router, the load bindables are id and parameters

This seems to be indeed the case:
```
@bindable public parameters?: Parameters;
...
@bindable public id?: string;
```

https://github.com/aurelia/aurelia/blob/master/packages/router/src/resources/load.ts#L17

**Comment by migajek** [comment]
Did a little debugging. 
The `LinkHandler` that does the actual `router.load` when the `a` is clicked, receives the following call:

![image](https://github.com/aurelia/aurelia/assets/581492/700d4210-47a0-45cb-923e-7ccb54145c2c)

This ends up with the following error
![image](https://github.com/aurelia/aurelia/assets/581492/739cefc1-acf4-4634-bc26-697ebc3e2bea)


However when I call the similar `load` directly from the controller code (so I use `<a click.trigger="myMethod(..` instead of `<a load="id: .., parameters.bind: ..`) it loads fine.

the only obvious difference, except for the call context, is the lack of `{ origin: target }` in my controller call.

Now, when I added the `{origin: $event.target}` to mimic the `LinkHandler` behavior, I'm getting the exact same exception.

But why? Well, that's beyond my understanding :)

**Comment by johnmanko** [comment]
Seems like it's been previously reported: https://github.com/aurelia/aurelia/issues/1835

**Comment by bigopon** [comment]
This is the log I get after clicking around in the recipes page:
![image](https://github.com/aurelia/aurelia/assets/9994529/50438f5f-b6c8-4f1b-bacc-1d380901a4f3)

Stackblitz is at https://stackblitz.com/edit/au2-conventions-vbk6gr

**Comment by migajek** [comment]
@bigopon I ran **yours** repro, headed to recipes and clicked **the underlined item**. This is what I ended up with:

![image](https://github.com/aurelia/aurelia/assets/581492/93b43906-ce2e-462e-990c-044250be096b)

please note I haven't touched anything.

I tried it in 3 browsers:
- Chrome 122
- Firefox 121
- Edge 122

all running Windows 10.
Also checked in incognito mode of Chrome and Firefox just to make sure.
Each time I'm getting the error as in the screenshot above.

What browser are you running?

**Comment by bigopon** [comment]
@migajek I clicked on the normal link, clicking on the underlined link gave me the same errors, thanks!

**Comment by bigopon** [comment]
So far, from what I see, this issue boils down to the `router.load` with the following differences:
1. Working:
    ```ts
    router.load({
      id: 'recipe-details',
      parameters: { recipeId: '52767' }
    })
    // or
    router.load({
      id: 'recipe-details',
      parameters: { recipeId: '52767' }
    }, {
      origin: undefined
    })
    ```
2. not working:
    ```ts
    router.load({
      id: 'recipe-details',
      parameters: { recipeId: '52767' }
    }, { 
      origin: the_anchor_with_href_to_the_recipe_page
    })
    ```

It means the presence of the `origin`, which is the `<a>` element, in the `router.load` options breaks it.
cc @jwx

**Comment by migajek** [comment]
following the discourse discussion - updated the repro to beta23
